app/data_providers/slate_providers/new_tab_slate_provider.py

```diff
@@ -0,0 +1,56 @@
+from typing import List
+
+from app.data_providers.slate_providers.slate_provider import SlateProvider
+from app.models.corpus_item_model import CorpusItemModel
+from app.models.localemodel import LocaleModel
+from app.data_providers.util import integer_hash
+from app.models.corpus_recommendation_model import CorpusRecommendationModel
+
+
+# Maximum tileId that Firefox can support. Firefox uses Javascript to store this value. The max value of a Javascript
+# number can be found using `Number.MAX_SAFE_INTEGER`. which is 2^53 - 1 because it uses a 64-bit IEEE 754 float.
+MAX_TILE_ID = 1 << 53 - 1
+# Generate tile_ids well out of the range of the MySQL-based system, which has a max tile_id of 99,999 as of 2023-03-13.
+# This is done to make it easy for engineers/analysts to see which system generated the identifier.
+MIN_TILE_ID = 100 * 100000
+
+
+class NewTabSlateProvider(SlateProvider):
```

Comment:
I'm curious about terminology here. Why are we calling newtab a slate? I thought slate is part of a recommendation surface, like topic or Popular. Are there plans to support multiple slates on new tab? Or is is just for compatibility with our current home abstractions?

Reply:
A slate is a ranked list of related recommendations. The [Pocket glossary](https://docs.google.com/document/d/17d-NKQCEbSqkY-mnN1PSwwf4xsHc5nl67WbjKRR88QY/edit#) adds:
> These items should be related such that the slate makes sense. Examples include: Trending items, Long reads, Business (topic).

A slate lineup is "an ordered list of Slates to be shown to users in a specific context." There are no plans to support multiple slates on New Tab. To avoid confusion, we don't return a `CorpusSlateLineup` to clients when they should only expect a single slate.